### PR TITLE
Feature/activity filter

### PR DESCRIPTION
**Pull request by cableman**
Missing the date filter

### CODE REVIEW

File: web/modules/custom/genlyd_maps/js/genlyd_maps_tabs.js [hunk 1]
```diff
@@ -3,40 +3,268 @@
...
+       *
+       * @param {string} key
+       *   Key for the value to add.
+       * @param {string} value
+       *   The value to add to the hash.
+       */
+      function addHashValue(key, value) {
+        var hash = window.location.hash;
+        hash += '&' + key + "=" + value + '&';
+        window.location.hash = hash.replace(/&{2,}/gi, '&');
```
Comment (tuj) [inline web/modules/custom/genlyd_maps/js/genlyd_maps_tabs.js]: Can you explain the two lines above?

File: web/modules/custom/genlyd_maps/js/genlyd_maps_tabs.js [hunk 2]
```diff
@@ -3,40 +3,268 @@
...
+       * @param {string} key
+       *   Key for the value to add.
+       * @param {string} value
+       *   The value to add to the hash.
+       */
+      function replaceHashValue(key, value) {
+        var hash = window.location.hash.substr(1);
+        var regex = new RegExp(key + '=\\w+', 'gi');
+        hash = hash.replace(regex, '');
+        hash = hash.replace(/&{2,}/, '&');
```
Comment (tuj) [inline web/modules/custom/genlyd_maps/js/genlyd_maps_tabs.js]: Explain, why is this necessary?

File: web/modules/custom/genlyd_maps/js/genlyd_maps_tabs.js [hunk 3]
```diff
@@ -3,40 +3,268 @@
...
+       * @param {string} key
+       *   Key for the value to remove.
+       * @param {string} value
+       *   The value to remove to the hash.
+       */
+      function removeHashValue(key, value) {
+        var hash = window.location.hash.substr(1);
+        var regex = new RegExp(key + '=' + value, 'gi');
+        hash = hash.replace(regex, '');
+        hash = hash.replace(/&{2,}/, '&');
```
Comment (tuj) [inline web/modules/custom/genlyd_maps/js/genlyd_maps_tabs.js]: Again explain

File: web/modules/custom/genlyd_maps/js/genlyd_maps_tabs.js [hunk 4]
```diff
@@ -3,40 +3,268 @@
...
+        // Ensures that the maps has all activities loaded.
+        if (viewsActivityFirstLoad && !showFilters) {
+          updateMap();
+        }
+
+        // Set first load to false.
+        viewsActivityFirstLoad = false;
+
+        // Negate show-filters as the filter function toggles it.
+        showFilters = !showFilters;
```
Comment (tuj) [inline web/modules/custom/genlyd_maps/js/genlyd_maps_tabs.js:85]: Can we avoid this? Sounds hackish...

File: web/modules/custom/genlyd_maps/src/Controller/ApiController.php [hunk 5]
```diff
@@ -24,23 +26,47 @@ class ApiController extends ControllerBase {
...
+
+    // Filter activities.
     $storage = \Drupal::entityTypeManager()->getStorage('node');
-    $ids = $storage->getQuery()
+    $query = $storage->getQuery()
       ->condition('type', 'activity')
-      ->condition('status', 1)
-      ->execute();
+      ->condition('status', 1);
+//      ->condition('field_data.value', $formatted, '<=');
```
Comment (tuj) [inline web/modules/custom/genlyd_maps/src/Controller/ApiController.php]: Maybe add as @TODO

**Review by tuj — CHANGES_REQUESTED** [review]

File: web/modules/custom/genlyd_maps/js/genlyd_maps_tabs.js (hunk 4 above)
Comment (cableman) [reply web/modules/custom/genlyd_maps/js/genlyd_maps_tabs.js:85]: No, it's a hack here or a bigger hack in the setFilter function.
